pkg/ccl/backupccl/show.go

```diff
@@ -420,62 +395,74 @@ you must pass the 'encryption_info_dir' parameter that points to the directory o
 			} else {
 				return err
 			}
-		} else if len(incLocations) > 0 {
-			// There was no error, so check for incrementals.
-			//
-			// If there are incrementals, find the backup paths and return them with the store.
-			// Otherwise, use the vacuous placeholders above.
+		}
+		mem := p.ExecCfg().RootMemoryMonitor.MakeBoundAccount()
+		defer mem.Close(ctx)
 
-			incStore, err = p.ExecCfg().DistSQLSrv.ExternalStorageFromURI(ctx, incLocations[0], p.User())
-			if err != nil {
+		var (
+			info        backupInfo
+			memReserved int64
+		)
+		info.subdir = computedSubdir
+
+		mkStore := p.ExecCfg().DistSQLSrv.ExternalStorageFromURI
+
+		info.defaultURIs, info.manifests, info.localityInfo, memReserved,
+			err = resolveBackupManifests(
+			ctx, &mem, baseStores, mkStore, fullyResolvedDest,
+			fullyResolvedIncrementalsDirectory, hlc.Timestamp{}, encryption, p.User())
+		defer func() {
+			mem.Shrink(ctx, memReserved)
+		}()
+		if err != nil {
+			if errors.Is(err, errLocalityDescriptor) && subdir == "" {
+				p.BufferClientNotice(ctx,
+					pgnotice.Newf("`SHOW BACKUP` using the old syntax ("+
+						"without the `IN` keyword) on a locality aware backup does not display or validate"+
+						" data specific to locality aware backups. "+
+						"Consider using the new `BACKUP INTO` syntax and `SHOW BACKUP"+
+						" FROM <backup> IN <collection>`"))
+			} else if errors.Is(err, cloud.ErrFileDoesNotExist) {
+				latestFileExists, errLatestFile := checkForLatestFileInCollection(ctx, baseStores[0])
+
+				if errLatestFile == nil && latestFileExists {
+					return errors.WithHintf(err, "The specified path is the root of a backup collection. "+
+						"Use SHOW BACKUPS IN with this path to list all the backup subdirectories in the"+
+						" collection. SHOW BACKUP can be used with any of these subdirectories to inspect a"+
+						" backup.")
+				}
+				return errors.CombineErrors(err, errLatestFile)
+			} else {
 				return err
 			}
-			incPaths, err = FindPriorBackups(ctx, incStore, IncludeManifest)
-			if err != nil {
-				return errors.Wrapf(err, "make incremental storage")
+		}
+		// If backup is locality aware, check that user passed at least some localities.
+
+		// TODO (msbutler): this is an extremely crude check that the user is
```

Comment:
I don't quite understand. This TODO is for master, and the warning is for 22.1 users. I don't want to incorporate ben's changes to show.go because they are not getting backported. Why should this "crude check + warning" get deleted?

Reply:
ahh sorry I thought Ben's change also touched show but it doesn't. We'd probably want to move https://github.com/cockroachdb/cockroach/pull/79135 to a helper or something.